src/osdc/Objecter.h

```diff
@@ -431,19 +435,28 @@ struct ObjectOperation {
   struct CB_ObjectOperation_cmpext {
     int* prval = nullptr;
     boost::system::error_code* ec = nullptr;
-    std::size_t* s = nullptr;
+    uint64_t* s = nullptr;
     explicit CB_ObjectOperation_cmpext(int *prval)
       : prval(prval) {}
-    CB_ObjectOperation_cmpext(boost::system::error_code* ec, std::size_t* s)
+    CB_ObjectOperation_cmpext(boost::system::error_code* ec, uint64_t* s)
       : ec(ec), s(s) {}
 
-    void operator()(boost::system::error_code ec, int r, const ceph::buffer::list&) {
+    void operator()(boost::system::error_code ec, int r,
+		    const ceph::buffer::list&) {
       if (prval)
         *prval = r;
-      if (this->ec)
-	*this->ec = ec;
+      if (this->ec) {
+	if (r < 0) {
+	  *this->ec = make_error_code(osd_errc::cmpext_failed);
```

Comment:
hi @idryomov;  are you really making a suggestion to rename variables *now*,  in a PR that was submitted for review in July of this year?

Reply:
Good  catch. Thank you.

Reply:
> hi @idryomov; are you really making a suggestion to rename variables _now_, in a PR that was submitted for review in July of this year?

Hi Matt,

Yes, I am.  First, I don't see how the "age" of the PR is relevant to the type of comments that are "OK" to be left by reviewers.  A particular reviewer might be seeing the PR for the first time, or it might be the first time they going over (a particular piece of) it in detail, or it might be that they just haven't noticed something before.

Second, for this particular PR, the branch kept getting pushed and test failures kept popping up (either due to new bugs introduced in this PR or pre-existing bugs getting exposed by this PR) in the interim.  The `cmpext` stuff is case in point: it wasn't present in the initial submission in July but showed up sometime in August, associated compare-and-write test failures in librbd were still outstanding in late September.  So here the date of the initial submission for review is even less irrelevant.

Third, bad naming, especially in a foundation library, tends to cause subtle, hard to notice, bugs.  Even in the PR itself, `cmpext_failed` error code was treated as "any cmpext failure" in `.h` file and as "cmpext op ran, data miscompared" in `.cc` file of the same class, most likely owning to the confusing name.  Would you expect a developer who is using this library to implement another library to not fall into the same trap?

And separately, I hope you agree that single-character names for API parameters are bad in general and that `s` for "offset from the start of the object to the first byte that mismatched" is bad in particular.

Reply:
Ok, so yes, that can be the case.  I agree that 's' is a poor substitute for "offset" or "off" or something else that expresses the intent of the variable.  Extra parens, rarely--typically that's helpful to less experienced c/c++ people, and is harmless even if it's a tic picked up from scheme or something.